lib/org/openpsa/sales/handler/view.php

```diff
@@ -103,6 +103,20 @@ private function _populate_toolbar()
         $this->bind_view_to_object($this->_salesproject);
     }
 
+    private function is_pdf_creatable()
+    {
+        $qb = org_openpsa_sales_salesproject_deliverable_dba::new_query_builder();
+        $qb->add_constraint('salesproject', '=', $this->_salesproject->id);
+        $qb->add_constraint('up', '=', 0);
+
+        if ($this->_salesproject->state != org_openpsa_sales_salesproject_dba::STATE_LOST) {
```

Comment:
The thing is, your constraint doesn't prevent creatings PDFs for lost salesprojects, what it does is that if the salesproject is lost, it will also add declined deliverables to the PDF. I think disabling the button for lost salesprojects makes sense, but the code as it currently is doesn't do that